File: back-end/Domain/BacklogItemServices/Commands/DTOs/BacklogRelationshipAction.cs
using Raven.Yabt.Database.Common.BacklogItem;
using Raven.Yabt.Domain.Common;

namespace Raven.Yabt.Domain.BacklogItemServices.Commands.DTOs;

public record BacklogRelationshipAction
{
	public string BacklogItemId { get; init; } = null!;
		
	public BacklogRelationshipType RelationType  { get; init; }
		
	public ListActionType ActionType  { get; init; }
}

File: front-end/src/app/core/api-models/backlog-item/item/BacklogRelationshipAction.ts
import { BacklogRelationshipType } from '@core/api-models/common/backlog-item';
import { ListActionType } from '@core/api-models/common/ListActionType';

export interface BacklogRelationshipAction {
	backlogItemId: string;
	relationType: keyof typeof BacklogRelationshipType;
	actionType: keyof typeof ListActionType;
}
